### ISSUE TITLE
Unable to insert boolean values.

### ISSUE THREAD

**Issue by lmwnshn** [cmu-db/terrier#1027]
Fixed in [this commit](https://github.com/cmu-db/terrier/pull/994/commits/51e53049a8bec7e2e7c5bd973fe4e7b35c7df514).

Before (main):
```
terrier=# create table foo (b bool);
CREATE TABLE
terrier=# insert into foo values ('t');
ERROR:  failed to convert string to another type, string to convert was t
terrier=# insert into foo values ('true');
ERROR:  failed to convert string to another type, string to convert was true
terrier=# insert into foo values (TRUE);
ERROR:  failed to convert string to another type, string to convert was t
terrier=# insert into foo values (1);
ERROR:  TryCastNumericAll not a numeric type! Desired type was BOOLEAN, number to convert was 1
terrier=# insert into foo values ('y');
ERROR:  failed to convert string to another type, string to convert was y
terrier=# insert into foo values ('yes');
ERROR:  failed to convert string to another type, string to convert was yes
terrier=# insert into foo values ('yes'::bool);
ERROR:  failed to convert string to another type, string to convert was yes
terrier=# insert into foo values ('true'::bool);
ERROR:  failed to convert string to another type, string to convert was true
```

After (TPL2):
```
terrier=# create table foo (a bool);
CREATE TABLE
terrier=# insert into foo values (TRUE), ('t'), ('true'), ('y'), ('yes'), ('on'), ('1');
INSERT 0 7
terrier=# insert into foo values (NULL);
INSERT 0 1
terrier=# insert into foo values (FALSE), ('f'), ('false'), ('n'), ('no'), ('off'), ('0');
INSERT 0 7
terrier=# select a,a,a from foo;
 a | a | a 
---+---+---
 t | t | t
 t | t | t
 t | t | t
 t | t | t
 t | t | t
 t | t | t
 t | t | t
   |   | 
 f | f | f
 f | f | f
 f | f | f
 f | f | f
 f | f | f
 f | f | f
 f | f | f
(15 rows)
```

**Comment by mbutrovich** [comment]
Pushed a slightly refactored commit:

https://github.com/cmu-db/terrier/pull/994/commits/bc3d1cfec829bd8b163d316247b54e47ad146f73

**Comment by lmwnshn** [comment]
Fixed by #994.
